### ISSUE TITLE
Error in SplitBrainResolver.PreStart when using ChannelTaskScheduler for internal-dispatcher

### ISSUE THREAD

**Issue by SchiessMax** [akkadotnet/akka.net#5962]
**Version Information**
Version of Akka.NET? 1.4.38
Which Akka.NET Modules? Akka.Cluster

**Describe the bug**
When using the Akka.Cluster.SBR.SplitBrainResolver while having configured the ChannelTaskScheduler as internal dispatcher an error is thrown in the PreStart method. The error is thrown when `Cluster.Get(Context.System)` is called.
I think this might be related to [#5861](https://github.com/akkadotnet/akka.net/pull/5861) 

The error that is logged:

```
ValueFactory attempted to access the Value property of this instance.
      [akka://CasActorSystem/system/cluster/core/daemon/downingProvider#1885816536]: Akka.Actor.ActorInitializationException: Exception during creation
       ---> System.InvalidOperationException: ValueFactory attempted to access the Value property of this instance.
         at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
         at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
         at System.Lazy`1.CreateValue()
         at System.Lazy`1.get_Value()
         at Akka.Actor.Internal.ActorSystemImpl.TryGetExtension[T](T& extension)
         at Akka.Actor.Internal.ActorSystemImpl.GetExtension[T]()
         at Akka.Actor.ActorSystemWithExtensions.WithExtension[T,TI](ActorSystem system)
         at Akka.Cluster.Cluster.Get(ActorSystem system)
         at Akka.Cluster.SBR.SplitBrainResolver.PreStart()
         at Akka.Actor.ActorBase.AroundPreStart()
         at Akka.Actor.ActorCell.<>c__DisplayClass176_0.<Create>b__0()
         at Akka.Actor.ActorCell.UseThreadContext(Action action)
         at Akka.Actor.ActorCell.Create(Exception failure)
         --- End of inner exception stack trace ---
         at Akka.Actor.ActorCell.Create(Exception failure)
         at Akka.Actor.ActorCell.SysMsgInvokeAll(EarliestFirstSystemMessageList messages, Int32 currentState) 
```

**To Reproduce**
1. Create an ActorSystem with the following configuration: 
```
akka {
    actor {
        provider = cluster
        default-dispatcher = {
            executor = "channel-executor"
            channel-executor.priority = "normal"
        }
       # Adding this part in combination with the SplitBrainResolverProvider causes the error
        internal-dispatcher = {
            executor = "channel-executor"
            channel-executor.priority = "high"
        }
    }
    remote {
        dot-netty.tcp {
            port = 5508
            hostname = 127.0.0.1
        }
        default-remote-dispatcher {
            executor = "channel-executor"
            channel-executor.priority = "high"
        }
        backoff-remote-dispatcher {
            executor = "channel-executor"
            channel-executor.priority = "low"
        }
    }
    cluster {
        seed-nodes = ["akka.tcp://ActorSystem@127.0.0.1:5508"]
        downing-provider-class = "Akka.Cluster.SBR.SplitBrainResolverProvider, Akka.Cluster"
    }
}
```

**Expected behavior**
The node should startup without errors

**Actual behavior**
An error is thrown in the SplitBrainResolver class

**Environment**
Are you running on Linux? Windows? Docker? Which version of .NET?
=> Error occurs in Linux, Windows and Docker. .NET Version: net6.0

**Comment by abeltsios** [comment]
Hello we have reached the same conclusion. (I was about to post the same bug here)

Some more context:

Split brain resolver calls _cluster = Cluster.Get(Context.System); in order to be able to subscribe to cluster events on PreStart phase.

That leads to exceptions being thrown during ActorSystems Extension's initialization code (ie, Lazy<object>.Value fails) when called from SplitBrainResolver.
As such SBR actor never initializes (as it is unsupervised).

The exception thrown is 
```
Cause: [akka://stx-pandora/system/cluster/core/daemon/downingProvider#2113384032]: Akka.Actor.ActorInitializationException: Exception during creation
 ---> System.InvalidOperationException: ValueFactory attempted to access the Value property of this instance.
   at System.Lazy`1.ViaFactory(LazyThreadSafetyMode mode)
   at System.Lazy`1.ExecutionAndPublication(LazyHelper executionAndPublication, Boolean useDefaultConstructor)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.get_Value()
   at Akka.Actor.Internal.ActorSystemImpl.TryGetExtension[T](T& extension)
   at Akka.Actor.Internal.ActorSystemImpl.GetExtension[T]()
   at Akka.Actor.ActorSystemWithExtensions.WithExtension[T,TI](ActorSystem system)
   at Akka.Cluster.Cluster.Get(ActorSystem system)
   at Akka.Cluster.SBR.SplitBrainResolver.PreStart()
   at Akka.Actor.ActorBase.AroundPreStart()
   at Akka.Actor.ActorCell.<>c__DisplayClass176_0.<Create>b__0()
   at Akka.Actor.ActorCell.UseThreadContext(Action action)
   at Akka.Actor.ActorCell.Create(Exception failure)
   --- End of inner exception stack trace ---
   at Akka.Actor.ActorCell.Create(Exception failure)
   at Akka.Actor.ActorCell.SysMsgInvokeAll(EarliestFirstSystemMessageList messages, Int32 currentState)
```



After a quick debug a simple change like the one bellow seems to resolve the issue (sorry for the git diff dump)

```
diff --git a/src/core/Akka.Cluster/SBR/SplitBrainResolver.cs b/src/core/Akka.Cluster/SBR/SplitBrainResolver.cs
index 12c203e2c..1a528e0f7 100644
--- a/src/core/Akka.Cluster/SBR/SplitBrainResolver.cs
+++ b/src/core/Akka.Cluster/SBR/SplitBrainResolver.cs
@@ -39,13 +39,21 @@ public static Props Props2(TimeSpan stableAfter, DowningStrategy strategy)
             return Props.Create(() => new SplitBrainResolver(stableAfter, strategy));
         }

+        protected override bool Receive(object message)
+        {
+            if (_cluster == null)
+            {
+                _cluster = Cluster.Get(Context.System);
+                _cluster.Subscribe(Self, InitialStateAsEvents, typeof(IClusterDomainEvent));
+            }
+            return base.Receive(message);
+        }
+
         // re-subscribe when restart
         protected override void PreStart()
         {
-            _cluster = Cluster.Get(Context.System);
-            _cluster.Subscribe(Self, InitialStateAsEvents, typeof(IClusterDomainEvent));
-
-            base.PreStart();
+           Self.Tell(new object());
+           base.PreStart();
         }
```

**Comment by Aaronontheweb** [comment]
So https://github.com/akkadotnet/akka.net/pull/5861#discussion_r866359711 was my attempted fix for this issue as Akka.NET v1.4.37 also has this bug, but it would occur in the constructor and blow up the cluster supervisor.

> As such SBR actor never initializes (as it is unsupervised).

__All actors are supervised__ - and in this case the SBR is supervised by the default supervision strategy (restart.)

The worst case scenario right now is the `SplitBrainResolver` should just restart, whereas before cluster initialization itself could potentially fail (because throwing in the constructor blows up the parent instantiating the actor too, which in this case was one of the core clustering actors.) While this issue is disconcerting I don't think it's going to have much production impact.

**Comment by SchiessMax** [comment]
Ok, so the SplitBrainResolver should still work. I will try it out, and if it does work I will to just have to live with the error log for now.

**Comment by Aaronontheweb** [comment]
It should definitely still work - but we'll fix this, either by breaking the SBR public API and passing in `Cluster` directly or by implementing some of @Zetanova 's suggestions here: https://github.com/akkadotnet/akka.net/discussions/5447

**Comment by abeltsios** [comment]
> > As such SBR actor never initializes (as it is unsupervised).
> 
> **All actors are supervised** - and in this case the SBR is supervised by the default supervision strategy (restart)

Pardon my phrasing! 
Noone triggers any restart and as such if someone tries to reproduce it given that he adds or removes nodes in a cluster, several deadletters can be observed

**Comment by Aaronontheweb** [comment]
Well I learned something new today: https://replit.com/join/jcsddyomuu-aaronontheweb

Throwing an exception in an actor's `PreStart` block will not allow the actor to be restarted: https://replit.com/join/jcsddyomuu-aaronontheweb

I'm trying to imagine if that's by design or not (it might be since the child actor can't really complete its startup sequence if it fails there.)

Either way, that means this bug is more severe than I thought. We'll get this patched right away.

**Comment by Aaronontheweb** [comment]
Have a fix for this in #5965

**Comment by Aaronontheweb** [comment]
Verifiable reproduction by @Arkatufus here https://github.com/akkadotnet/akka.net/pull/5966

**Comment by Aaronontheweb** [comment]
Resolved cia #5965
